backend/server/src/client.rs

```diff
@@ -6,18 +6,30 @@ use tokio::{
     net::{unix::SocketAddr, UnixListener, UnixStream},
 };
 use tracing::{debug, info, warn};
+use std::sync::mpsc::Sender;
 
 pub struct UnixService {
     listener: UnixListener,
+    /// Channel to emit drvs to be evaluated
+    dispatch: DispatchChannels,
+}
+
+/// Channels which can be used to communicate actions to other services
+#[derive(Clone)]
+struct DispatchChannels {
+    eval_sender: Sender<String>,
 }
 
 impl UnixService {
-    pub async fn bind_to_path(socket_path: &Path) -> Result<Self> {
+    // TODO: We should probably use a builder pattern to pass eval channel and other items
```

Comment:
Is there a better way?

Reply:
Maybe something like:

```rust
struct UnixServiceBuilder {
    dispatch: DispatchChannels,
}

impl UnixServiceBuilder {
    async fn bind_to_path(...) -> Result<UnixService> {
        ...
    }
}

impl UnixService {
    fn with_dispatch(dispatch: ...) -> UnixServiceBuilder {
        UnixServiceBuilder { dispatch }
    }
}
```

Reply:
Think https://docs.rs/tonic/latest/src/tonic/transport/channel/endpoint.rs.html#28-51 would be good?
```
impl UnixServiceBuilder {
    fn with_dispatch(self, dispatch: DispatchChannels) -> Self {
        Endpoint {
            dispatch: Some(dispatch),
            ..self
        }
    }
}
```
This is probably over optimizing for now. but eventually the unix service.

Part of me is feeling like since this is an internal service, and we will likely do most of the configuration determination when resolving config. So a builder is less desirable, as we don't really have a "few mandatory things and a bunch of optional things", instead we just have "a few mandatory things".